src/core_modules/capture-core/metaDataStoreLoaders/programs/loadPrograms.js

```diff
@@ -150,36 +78,32 @@ function getCategories(missingPrograms) {
         }, []) : [];
 }
 
-export default async function loadProgramsData(storageController: StorageController, stores: Object) {
-    const metaPrograms = await metaProgramsSpec.get();
-    const optionSetsMeta = getOptionSetsMeta(metaPrograms);
-    const missingPrograms = await getMissingPrograms(
-        metaPrograms,
-        storageController,
-        stores[programsStoresKeys.PROGRAMS],
-    );
+async function loadProgramBatch(programBatch) {
+    const programIds = programBatch.map(program => program.id);
+    const { convertedData: programs } = await storePrograms(programIds);
+    await storeProgramRules(programIds);
+    await storeProgramRulesVariables(programIds);
+    await storeProgramIndicators(programIds);
+    return programs;
+}
+
+export async function loadPrograms() {
+    const programsOutline = await queryProgramsOutline();
+    const optionSetsOutline = getOptionSetsOutline(programsOutline);
+    const missingPrograms = await getMissingPrograms(programsOutline);
 
-    const programBatches = chunk(missingPrograms, batchSize);
+    const programBatches = chunk(missingPrograms, 50);
```

Comment:
I am think out loud here, where is 50 coming from? Is it scalable?

Reply:
I didn't change it, it's something we settled on a long time ago.